Stratis.Bitcoin/Features/MemoryPool/MempoolManager.cs

```diff
@@ -9,23 +9,49 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool scheduler.
+    /// </summary>
     public class MempoolScheduler : AsyncLock
```

Comment:
This is unfortunate naming, this `MempoolScheduler` is in fact just a way to provide exclusivity for operation (something like lock) and only internally it uses a scheduler. But from the memory pool point of view it is not a scheduler. This should probably be renamed to `MempoolLock` or something like that.

This PR is about the comment, so the comment here is somehow misleading because it should rather say that it is a lock not scheduler.